src/main/scala/beam/agentsim/agents/ridehail/RideHailVehicleManager.scala

```diff
@@ -173,7 +173,7 @@ class RideHailVehicleManager(val rideHailManager: RideHailManager, boundingBox:
   }
 
   def getIdleVehiclesAndFilterOutExluded: mutable.HashMap[Id[Vehicle], RideHailAgentLocation] = {
-    idleRideHailVehicles.filterNot(elem => rideHailManager.doNotUseInAllocation.contains(elem._1))
+    idleRideHailVehicles.filter(elem => !rideHailManager.doNotUseInAllocation.contains(elem._1))
```

Comment:
Is this preference, or does this actually make a difference in performance?

Reply:
Yes, it is, here is what I saw in the profiler before optimization (overall)
![image](https://user-images.githubusercontent.com/5107562/67788770-7d491480-faa5-11e9-9ea1-3b047dd94af5.png)

Only RHM thread (before optimization):
![image](https://user-images.githubusercontent.com/5107562/67788823-9225a800-faa5-11e9-9ef0-53e59a35a813.png)

Only RHM thread (after optimization):
![image](https://user-images.githubusercontent.com/5107562/67789321-5d662080-faa6-11e9-8f46-b52b86109817.png)

Reply:
Odd...I'd have to dig in - but will defer to the benchmark. I wonder if the new collections API will make this moot